mysql/data.go

```diff
@@ -0,0 +1,241 @@
+	f, err := strconv.ParseFloat(val, 64)
+	if err != nil {
+		return f, fmt.Errorf("can't convert to float64: %w", err)
+	}
+	return f, err
+}
+
+func convInt64(val string) (int64, error) {
+	i, err := strconv.ParseInt(val, 10, 64)
+	if err != nil {
+		return i, fmt.Errorf("can't convert to int64: %w", err)
+	}
+	return i, err
+}
+
+// convTimestamp maps a source DB timestamp into a go Time Spanner timestamp
+// It handles both datetime and timestamp conversions.
+func convTimestamp(srcTypeName string, TimezoneOffset string, val string) (t time.Time, err error) {
+	//Mysql dump outputs time in "2006-01-02 15:04:05" format
+	//we consider timezone for  timestamp type, for datetime type it is considered UTC
+	if srcTypeName == "timestamp" {
+		if TimezoneOffset == "" {
+			TimezoneOffset = "+00:00" // if timezone is not specified in mysqldump, we consider UTC time
+		}
+		time_new := strings.Split(val, " ") // convert time from format "2006-01-02 15:04:05" to "2006-01-02T15:04:05+00:00"
+		time_joined := strings.Join(time_new, "T")
+		time_joined = time_joined + "" + TimezoneOffset
+		t, err = time.Parse(time.RFC3339, time_joined)
+	} else {
+		t, err = time.Parse("2006-01-02 15:04:05", val)
+	}
+	if err != nil {
+		return t, fmt.Errorf("can't convert to timestamp (mysql type: %s)", srcTypeName)
+	}
+	return t, err
+}
+
+// convArray converts a source database string value (representing an
+// array) to an appropriate Spanner array value. It is the caller's
+// responsibility to detect and handle the case where the entire array
+// is NULL. However, convArray does handle the case where individual
+// array elements are NULL. In other words, convArray handles "{1,
+// NULL, 2}", but it does not handle "NULL" (it returns error).
+func convArray(spannerType ddl.ScalarType, srcTypeName string, v string) (interface{}, error) {
+	v = strings.TrimSpace(v)
+	// Handle empty array. Note that we use an empty NullString array
+	// for all Spanner array types since this will be converted to the
+	// appropriate type by the Spanner client.
+	if v == "" {
+		return []spanner.NullString{}, nil
+	}
+
+	a := strings.Split(v, ",")
+
+	// The Spanner client for go does not accept []interface{} for arrays.
+	// Instead it only accepts slices of a specific type eg : []string
+	// Hence we have to do the following case analysis
+	// NOTE : Mysql only support array<string>
+	switch spannerType.(type) {
+
+	case ddl.String:
+		var r []spanner.NullString
+		for _, s := range a {
+			if s == "NULL" {
+				r = append(r, spanner.NullString{Valid: false})
+				continue
+			}
+			s, err := processQuote(s)
+			if err != nil {
+				return []spanner.NullString{}, err
+			}
+			r = append(r, spanner.NullString{StringVal: s, Valid: true})
+		}
+		return r, nil
+	}
+	return []interface{}{}, fmt.Errorf("array type conversion not implemented for type %v", reflect.TypeOf(spannerType))
+}
+
+// processQuote returns the unquoted version of s.
```

Comment:
Please format this comment well. There are some issues, like extra spaces between two sentences. The line length should be 80 chars.